tests/test_bad_eventlog.py

```diff
@@ -3,46 +3,36 @@
 from pathlib import Path
 from zipfile import ZipFile
 
-from spark_log_parser.eventlog import EventLogBuilder
+from spark_log_parser import eventlog, extractor
 
 
 class BadEventLog(unittest.TestCase):
-    def test_multiple_context_ids(self):
-        event_log = Path("tests", "logs", "bad", "non-unique-context-id.zip").resolve()
+    def check_value_error(self, event_log_path, msg):
 
         with tempfile.TemporaryDirectory() as temp_dir:
-            with self.assertRaises(
-                ValueError, msg="Not all rollover files have the same Spark context ID"
-            ):
-                EventLogBuilder(event_log.as_uri(), temp_dir).build()
+            with self.assertRaises(ValueError, msg=msg):
+                event_log_paths = extractor.Extractor(event_log_path.as_uri(), temp_dir).extract()
+                eventlog.EventLogBuilder(event_log_paths, temp_dir).build()
```

Comment:
Nice abstraction. I made an error when I coded the first version of this thinking that assertRaises will assert that the value of its "msg" argument matches that of the attribute of the exception. That's not the case: https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaises

I should've done,
```
with self.assertRaises(ValueError) as cm:
                event_log_paths = extractor.Extractor(event_log_path.as_uri(), temp_dir).extract()
                eventlog.EventLogBuilder(event_log_paths, temp_dir).build()
assert str(cm.exception) == msg, "Exception message matches"
```

Reply:
Done!  Had to fix some of the error messages to match, too.

btw, curious why you went went unittest here instead of using pytest's exception testing capabilities.

Reply:
I don't recall having made a deliberate choice between the 2, but I'm interested in differences that may exist between the 2 or examples with the pytest alternative. I do like that these related tests are grouped in a class, and that class offers `assertRaises`... felt right

Reply:
I think I just Googled "python test raise exception": https://stackoverflow.com/questions/129507/how-do-you-test-that-a-python-function-throws-an-exception